### ISSUE TITLE
protobufs upgrade fails

### ISSUE THREAD

**Issue by Noah-Huppert** [void-linux/void-packages#24344]
### System

* xuname:  
  *output of ``xuname`` (part of xtools)*
```
Void 5.4.50_1 x86_64 GenuineIntel notuptodate rrrmFFFFFFFFFFF
```
* package:  
  *affected package(s) including the version*: ``xbps-query -p pkgver <pkgname>``

```
protobuf24-3.13.0_1
protobuf23-3.12.4_1
protobuf-3.13.0_1
```

### Expected behavior

Upgrading protobufs (when upgrading my system via `xbps-install -Syu`) should not fail due to a newer version of protobufs providing the same binary which the older version provides.

### Actual behavior

```
% xbps-install -Syu
...
protobuf24-3.13.0_1: collecting files...
protobuf-3.13.0_1: collecting files...
protobuf23-3.12.4_1: collecting files...
ERROR: protobuf23-3.12.4_1: file `/usr/bin/protoc' already installed by package protobuf2
4-3.13.0_1.
Transaction failed! see above for errors.
```

[full output](https://gist.github.com/Noah-Huppert/b010256dbb6b8b31a402d7db9c84fedb)

It seems like it fails to upgrade from protobuf23 to protobuf24 bc they provide the same binary. 

### Steps to reproduce the behavior

1. Have `protobuf23` installed. 
2. Try to upgrade to `protobuf24`

**Comment by Noah-Huppert** [comment]
Running `xbps-install -Syu protobuf` after the failure does finish the upgrade successfully.

**Comment by sgn** [comment]
`protobuf24-3.13.0_2` conflicts with `protobuf23`.

**Comment by Noah-Huppert** [comment]
Hi Danh,

Would you mind explaining in a bit more detail? To me it seems like an
upgrade between two incremental versions shouldn't fail bc one version has
the same binary as the other version.

**Comment by sgn** [comment]
Yes, it shouldn't fail. Hence, we should have `protobuf24` `conflicts` with `protobuf23` and `prorovides` `protobuf23`.

The original change forgot to do it, and it's fixed in c260514218 (protobuf24: update conflicts list, 2020-08-18)

**Comment by Noah-Huppert** [comment]
Thanks! I understand now.
